src/ServiceControl.Audit/Auditing/AuditIngestionComponent.cs

```diff
@@ -57,16 +57,21 @@ AuditIngestionCustomCheck.State ingestionState
             ingestor = new AuditIngestor(auditPersister, settings);
 
             var ingestion = new AuditIngestion(
-                async (messageContext, dispatcher) =>
-                {
-                    var taskCompletionSource = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
-                    messageContext.SetTaskCompletionSource(taskCompletionSource);
+                 (messageContext, dispatcher) =>
+                 {
+                     async Task Process()
+                     {
+                         var taskCompletionSource = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
+                         messageContext.SetTaskCompletionSource(taskCompletionSource);
+
+                         receivedMeter.Mark();
 
-                    receivedMeter.Mark();
+                         await channel.Writer.WriteAsync(messageContext).ConfigureAwait(false);
+                         await taskCompletionSource.Task.ConfigureAwait(false);
+                     }
 
-                    await channel.Writer.WriteAsync(messageContext).ConfigureAwait(false);
-                    await taskCompletionSource.Task.ConfigureAwait(false);
-                },
+                     return settings.OnMessage(messageContext.MessageId, messageContext.Headers, messageContext.Body, Process);
```

Comment:
I don't remember the details but I thought there were some delegate caching issues when using method groups. Might be worth to double check. (https://github.com/dotnet/roslyn/issues/5835)

Reply:
The message filtering logic has been refactored and we no longer use a method group.